### PR TITLE
Added file source post=moveaway

### PR DESCRIPTION
**Pull request by margirova**
Added new post keyword "moveaway", the code puts processed files into specified folder.
- tested
- ready to merge, if everything is fine

### CODE REVIEW

**Comment by margirova** [comment]
fixed

File: bspump/file/fileabcsource.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ class FileABCSource(TriggerSource):
 		'post': 'move', # one of 'delete', 'noop' and 'move'
 		'exclude': '', # glob of filenames that should be excluded (has precedence over 'include')
 		'include': '', # glob of filenames that should be included
-		'encoding': ''
+		'encoding': '',
+		'path_processed': ''
```
Comment (ateska) [inline bspump/file/fileabcsource.py]: `move_destination`

File: bspump/file/fileabcsource.py [hunk 2]
```diff
@@ -43,9 +44,17 @@ def __init__(self, app, pipeline, id=None, config=None):
 		self.include = self.Config['include']
 		self.exclude = self.Config['exclude']
 		self.encoding = self.Config['encoding']
+		
+		self.PathProcessed = self.Config['path_processed']
 
-		metrics_service = app.get_service('asab.MetricsService')
+		if (self.PathProcessed != ''):
+			if (self.post == 'move') and (not os.path.isdir(self.PathProcessed)):
+				os.mkdir(self.PathProcessed)
```
Comment (ateska) [inline bspump/file/fileabcsource.py]: `os.makedirs`

File: bspump/file/fileabcsource.py [hunk 3]
```diff
@@ -141,7 +150,13 @@ def __init__(self, app, pipeline, id=None, config=None):
 			elif self.post == "noop":
 				os.rename(locked_filename, filename)
 			else:
-				os.rename(locked_filename, filename + '-processed')
+				new_filename = filename + '-processed'
+				os.rename(locked_filename, new_filename)
```
Comment (ateska) [inline bspump/file/fileabcsource.py]: No double renaming, please

File: bspump/file/globscan.py [hunk 4]
```diff
@@ -15,7 +22,7 @@ def _is_file_open(fname):
 		return len(result.stdout) != 0
 
 
-def _glob_scan(path, gauge, loop, exclude='', include=''):
+def _glob_scan(path, gauge, loop, post, exclude='', include='', path_processed=None):
```
Comment (ateska) [inline bspump/file/globscan.py]: You don't need `path_processed` here - just don't count processed files.
I would even think of removing that metric completely - we cannot calculate that correctly.

**Pull request merged**